core/vm/jump_table.go

```diff
@@ -47,13 +47,32 @@ type operation struct {
 	// jumps indicates whether operation made a jump. This prevents the program
 	// counter from further incrementing.
 	jumps bool
+	// writes determines whether this a state modifying operation
+	writes bool
```

Comment:
I'm not sure what's the ultimate intention of `writes` , but you could argue that `CALL` is also a state modifying operation, since if you send ether to a non-existing account, a new account is created in the state. You have not added the `writes`-flag to CALL (and friends) though ?